### ISSUE TITLE
Support repository-azure snapshot plugin with "managed identity"

### ISSUE THREAD

**Issue by Leaf-Lin** [elastic/elasticsearch#89633]
### Description

Microsoft is recommending Managed Identities as a superior way of authorizing access.  
- https://docs.microsoft.com/en-us/azure/storage/blobs/authorize-managed-identity
- https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview

A common challenge for developers is the management of secrets, credentials, certificates, and keys used to secure communication between services. Managed identities eliminate the need for developers to manage these credentials.

Azure Blob Storage supports Azure Active Directory (Azure AD) authentication with [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview). Managed identities for Azure resources can authorize access to blob data using Azure AD credentials from applications running in Azure virtual machines (VMs), function apps, virtual machine scale sets, and other services. By using managed identities for Azure resources together with Azure AD authentication, you can avoid storing credentials with your applications that run in the cloud.

**Comment by artificial-aidan** [comment]
I put together a POC on a fork [here](https://github.com/artificialinc/elasticsearch/tree/aidan/8-10-0-default-azure-credential).

First time working in Java, and this repo is pretty complicated, but it's working for me. Maybe somebody could take over who is more familiar, or help me work through some of the non-code changes needed.

I think it could be done more cleanly but adding new settings and whatnot, but as a first time contribution, that wasn't as well documented at first glance.

**Comment by artificial-aidan** [comment]
Bump? Any thoughts, should I open a PR?

**Comment by horvatal** [comment]
I would try a PR, than it gets more visible to the maintainers ;-)

**Comment by sabarishkre** [comment]
@artificial-aidan Any PR out yet?

**Comment by artificial-aidan** [comment]
No, but I've been running the code in production since I made this issue, no problems.

**Comment by sabarishkre** [comment]
You mean you've been running a build from your branch, containing your changes to support workload identity in production? 
If yes, can you provide some more details - does it work with AKS having pod identities as well? Thanks!

**Comment by artificial-aidan** [comment]
> You mean you've been running a build from your repo/branch, containing your changes to support workload identity, in production? 
> If yes, can you provide some more details - does it work with AKS having pod identities as well? Thanks!

I do not use pod identities, but it might work. It just uses the default methods of loading credentials. For workload identities it works fine.

**Comment by bcully** [comment]
I believe this issue should have been resolved with #111346
